File: golang/easy/compare-triplets.go
//HackerRank

package main

import (
	"fmt"
)

func compareTriplets(a []int, b []int) []int {
	alicePoints := 0
	bobPoints := 0

	// Compare elements of both arrays
	for i := 0; i < len(a); i++ {
		if a[i] > b[i] {
			alicePoints++
		} else if a[i] < b[i] {
			bobPoints++
		}
	}

	// Return results as a slice
	return []int{alicePoints, bobPoints}
}

func main() {
	// Example inputs
	a := []int{5, 6, 7}
	b := []int{3, 6, 10}

	// Call the function and print the result
	result := compareTriplets(a, b)
	fmt.Println(result) // Output: [1 1]
}

File: easy/typescript/compare-triplets.ts

//HackerRank

function compareTriplets(a: number[], b: number[]): number[] {
  let bobPoints = 0
  let alicePoints = 0
  
  for(let i = 0; i < a.length; i++){
      if(a[i] > b[i]){
          alicePoints++
      }
      
      if(a[i] < b[i]){
          bobPoints++
      }
  }
  
  return [alicePoints, bobPoints]
}